### PR TITLE
Adding static website example for azure-nextgen

### PR DESCRIPTION
**Pull request by viveklak**

### CODE REVIEW

File: azure-nextgen-static-website/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        "pretty": true,
+        "noFallthroughCasesInSwitch": true,
+        "noImplicitAny": true,
+        "noImplicitReturns": true,
+        "forceConsistentCasingInFileNames": true,
+        "strictNullChecks": true
+    },
+    "files": [
+        "index.ts",
+        "staticWebsite.ts"
```
Comment (mikhailshilkov) [inline azure-nextgen-static-website/tsconfig.json:18]: What's `staticWebsite.ts`?

File: azure-nextgen-static-website/index.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    origins: [{
+        hostName: endpointOrigin,
+        httpPort: 80,
+        httpsPort: 443,
+        name: pulumi.interpolate `${cdnEndpointName}-origin-${randomSuffix.result}`,
+    }],
+    profileName: profile.name,
+    queryStringCachingBehavior: "NotSet",
+    resourceGroupName: resourceGroup.name,
+    tags: {},
```
Comment (mikhailshilkov) [inline azure-nextgen-static-website/index.ts:82]: I think we can remove a bunch of irrelevant properties

File: azure-nextgen-static-website/index.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+const endpoint = new cdn.Endpoint("CDNEndpoint", {
+    contentTypesToCompress: [],
+    endpointName: cdnEndpointName,
+    isCompressionEnabled: false,
+    isHttpAllowed: true,
+    isHttpsAllowed: true,
+    originHostHeader: endpointOrigin,
+    origins: [{
+        hostName: endpointOrigin,
+        httpPort: 80,
```
Comment (mikhailshilkov) [inline azure-nextgen-static-website/index.ts:75]: It looks like only HTTPS is enabled for the blob storage

File: azure-nextgen-static-website/index.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+const config = new pulumi.Config();
+const storageAccountName = config.get("storageAccountName") || pulumi.interpolate `site${randomSuffix.result}`;
+const cdnEndpointName = config.get("cdnEndpointName") || pulumi.interpolate `cdn-endpnt-${storageAccountName}`;
+const cdnProfileName = config.get("cdnProfileName") || pulumi.interpolate `cdn-profile-${storageAccountName}`;
+
+const profile = new cdn.Profile("profileResource", {
+    profileName: cdnProfileName,
+    resourceGroupName: resourceGroup.name,
+    sku: {
+        name: "Premium_Verizon",
```
Comment (mikhailshilkov) [inline azure-nextgen-static-website/index.ts:30]: I'd use `Standard_Microsoft` instead. Btw, let's use enums where they are defined.

File: azure-nextgen-static-website/index.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+// TODO: Remove after autonaming support is added.
+const randomSuffix = new random.RandomString("random", {
+    length: 10,
+    special: false,
+    upper: false,
+});
+
+const resourceGroup = new resources.ResourceGroup("resourceGroup", {
+    resourceGroupName: pulumi.interpolate `rg${randomSuffix.result}`,
+    location: "westus",
```
Comment (mikhailshilkov) [inline azure-nextgen-static-website/index.ts:18]: No need to hard-code the location

File: azure-nextgen-static-website/index.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        bypass: "AzureServices",
+        defaultAction: "Allow",
+        ipRules: [],
+        virtualNetworkRules: [],
+    },
+    resourceGroupName: resourceGroup.name,
+    sku: {
+        name: "Standard_LRS",
+    },
+    tags: {},
```
Comment (mikhailshilkov) [inline azure-nextgen-static-website/index.ts:61]: I think we can remove a bunch of irrelevant properties

**Review by mikhailshilkov — COMMENTED** [review]
Looks good - let's clean up a bit and ship it

File: azure-nextgen-static-website/tsconfig.json (hunk 1 above)
Comment (viveklak) [reply azure-nextgen-static-website/tsconfig.json]: Argh - I copied the azure version and fixed it up. Its the component resource we used to use for the azure provider before the provider added static website support. Removed.

File: azure-nextgen-static-website/index.ts (hunk 2 above)
Comment (viveklak) [reply azure-nextgen-static-website/index.ts]: Might have to do with the arm template I started with. Cleaned up to a large extent now.

File: azure-nextgen-static-website/index.ts (hunk 3 above)
Comment (viveklak) [reply azure-nextgen-static-website/index.ts]: Left over from arm template. Fixed.

File: azure-nextgen-static-website/index.ts (hunk 4 above)
Comment (viveklak) [reply azure-nextgen-static-website/index.ts]: Done.

File: azure-nextgen-static-website/index.ts (hunk 5 above)
Comment (viveklak) [reply azure-nextgen-static-website/index.ts]: If I exclude resourceGroup creation fails. Switched it to a config entry with a default instead.

File: azure-nextgen-static-website/index.ts (hunk 6 above)
Comment (viveklak) [reply azure-nextgen-static-website/index.ts]: Done.

File: azure-nextgen-static-website/README.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+## Running the App
+
+1.  Create a new stack:
+
+    ```
+    $ pulumi stack init dev
+    ```
+
+1.  Login to Azure CLI (you will be prompted to do this during deployment if you forget this step):
```
Comment (lblackstone) [inline azure-nextgen-static-website/README.md:19]: May want to link to info about not sending creds to our servers again here.

File: azure-nextgen-static-website/index.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+// TODO: Remove after autonaming support is added.
+const randomSuffix = new random.RandomString("random", {
+    length: 10,
+    special: false,
+    upper: false,
+});
+
+const config = new pulumi.Config();
+const location = config.get("location") || "westus";
```
Comment (lblackstone) [inline azure-nextgen-static-website/index.ts:17]: Do we have an enum we could use here?

**Review by lblackstone — APPROVED** [review]

File: azure-nextgen-static-website/index.ts (hunk 8 above)
Comment (viveklak) [reply azure-nextgen-static-website/index.ts:17]: I didn't see enums for location - I know the TF azure provider SDK had them...

Comment (mikhailshilkov) [reply azure-nextgen-static-website/index.ts:17]: Azure doesn't have them. TF ones are our overlays.

**Pull request merged**